Fixes to document structure in Express.js migration guide (#1365)
encoredev/encore: docs/how-to/express-migration.md

| [Request validation](/blog/event-loops)                     |           [Cron Jobs](/docs/ts/primitives/cron-jobs)           |             [Local tracing](/docs/observability/tracing) |
| [Local Development Dashboard](/docs/observability/dev-dash) |

## Migration guide

> We have outlined two different strategies you can use to migrate your existing Express.js application to Encore.ts. Pick the strategy that suites your situation and application the the best.
> 

Reviewer: ```suggestion
Below we've outlined two main strategies you can use to migrate your existing Express.js application to Encore.ts. Pick the strategy that best suits your situation and application.

```

<Accordion>

### Forklift migration (quick start)

> You can use a forklift migration strategy and move the entire application over to Encore.ts in one shot by wrapping your existing HTTP router in a catch-all handler.

Reviewer: ```suggestion
When you quickly want to migrate to Encore.ts and don't need all the functionality to begin with, you can use a forklift migration strategy. This approach moves the entire application over to Encore.ts in one shot, by wrapping your existing HTTP router in a catch-all handler.
```

**Approach benefits**

- You can get your application up and running with Encore.ts quickly and start moving features over to Encore.ts while the rest of the application is still untouched.

> - You will see a performance boost right away because the HTTP layer is now running on the Encore Rust runtime. But to get the full performance benefits, you will need to start using Encore's [API declarations](/docs/ts/primitives/services-and-apis) and [infrastructure declarations](/docs/ts#explore-how-to-use-each-backend-primitive).
> 

Reviewer: ```suggestion
- You will see a partial performance boost right away because the HTTP layer is now running on the Encore Rust runtime. But to get the full performance benefits, you will need to start using Encore's [API declarations](/docs/ts/primitives/services-and-apis) and [infrastructure declarations](/docs/ts#explore-how-to-use-each-backend-primitive).

```

);
```

By mounting your existing app router in this way, it will work as a catch-all handler for all HTTP requests and responses.

#### 7. Run you app locally

You will now be able to run your Express.js app locally using the `encore run` command.

#### Next steps: Incrementally move over Encore.ts to get all the benefits

You can now gradually break out specific endpoints using the Encore's [API declarations](#apis) and introduce infrastructure declarations for databases and cron jobs etc. This will let Encore.ts understand your application and unlock all Encore.ts features. See the [Feature-by-feature migration](#feature-by-feature-migration) section for more details. You will eventually be able to remove Express.js as a dependency and run your app entirely on Encore.ts.

For more thoughts on migrating an existing backend to Encore.ts, check out our [general migration guide](/docs/how-to/migrate-to-encore). You can also [join Discord](https://encore.dev/discord) to ask questions and meet fellow Encore developers.

#### Forklift example

<div className="not-prose my-10">
   <Editor projectName="expressForklift" />
</div>

</Accordion>

<Accordion>

### Full migration

> This approach aims to replace your apps dependency on Express.js with Encore.ts. Below you find two examples that you can use to identify the refactoring you will need to do. In the next section you will find a [Feature-by-feature migration](#feature-by-feature-migration) guide to help you with the refactoring details.

Reviewer: ```suggestion
This approach aims to fully replace your applications dependency on Express.js with Encore.ts, unlocking all the features and performance of Encore.ts.

Below are two examples that you can use to identify the refactoring you will need to do. In the next section you will find a [Feature-by-feature migration](#feature-by-feature-migration) guide to help you with the refactoring details.
```

**Approach benefits**

> - You will get all the advantages of Encore.ts, like [distributed tracing](/docs/observability/tracing) and [architecture diagrams](/docs/observability/encore-flow), which rely on the [Encore application model](https://encore.dev/docs/introduction#meet-the-encore-application-model).
> - Get the [full performance benefit](https://encore.dev/blog/event-loops) of Encore.ts - **9x faster** than Express.js.

Reviewer: ```suggestion
- Get all the advantages of Encore.ts, like [distributed tracing](/docs/observability/tracing) and [architecture diagrams](/docs/observability/encore-flow), which rely on the [Encore application model](https://encore.dev/docs/introduction#meet-the-encore-application-model).
- Get the [full performance benefit](https://encore.dev/blog/event-loops) of Encore.ts - **9x faster** than Express.js.
```
